### PR TITLE
Refactor Tabbed Panel Component & Linode Details Navigation

### PR DESCRIPTION
**Pull request by WilkinsKa1**
## Description

Leveraging reach tabs and reach-router for TabbedPanel and creating new pattern for routed tab navigation (starting with LinodeDetailsNavigation). Please note this work will be merged and isolated to a feature branch until all tab navigation patterns have been updated.

## Type of Change
- Non breaking change ('update')

## Note to Reviewers
See todo list :) 

## Todo:
- [ ] Firefox focus issues (link pattern only- I believe this is due to FF wanting the links to be buttons, not `a`, which is a side effect of using `Link` (https://stackoverflow.com/questions/11704828/how-to-allow-keyboard-focus-of-links-in-firefox/11713537#11713537))
- [x] Tabs not changing on arrow key events (router pattern only)
- [x] CSS loading only once
- [x] Re-add lazy loading (see PR comments)
- [x] TS linting errors
- [x] Documentation

### CODE REVIEW

File: packages/manager/src/components/core/ReachTabPanel.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import { TabPanel } from '@reach/tabs';
+import '@reach/tabs/styles.css';
```
Comment (Jskobos) [inline packages/manager/src/components/core/ReachTabPanel.ts:2]: If this is just normal CSS maybe we could import it once at the app root level?

File: packages/manager/src/features/linodes/LinodesDetail/LinodesDetailNavigation.tsx [hunk 2]
```diff
@@ -1,58 +1,23 @@
...
-});
-
-const LinodeAdvanced = DefaultLoader({
-  loader: () => import('./LinodeAdvanced')
-});
-
-const LinodeBackup = DefaultLoader({
-  loader: () => import('./LinodeBackup')
-});
+import LinodeActivity from './LinodeActivity';
```
Comment (Jskobos) [inline packages/manager/src/features/linodes/LinodesDetail/LinodesDetailNavigation.tsx:12]: This loses the lazy loading, which Dajahi just fixed up to use React.lazy. I think if you wrapped the Reach `<Router>` component in a SuspenseLoader component, you could re-add the lazy stuff. In a section of the app this big I think the lazy loading is helpful to keep chunks small.

**Comment by DevDW** [comment]
Aside from the points Jared raised, this is looking good to me so far. I tried tinkering a bit to get `LinodeSettings` to work, but wasn't able to without replacing the `component` attribute with `as`, as with the other tabs.

**Comment by johnwcallahan** [comment]
> I remember in the demo you gave the tabs would change content when arrow-keying across them. Was removing that a tradeoff we decided on? (It was pretty cool)

I was also wondering about this. Does using the router of something to do with it? Looking good so far, though.

**Comment by WilkinsKa1** [comment]
@Jskobos @DevDW @johnwcallahan thanks for the reviews! 

This should be good to go now. I fixed the linked content arrow key tabbing issue, which ended up also fixing the TS errors! I added back in lazy loading and am now only loading the css for Reach once in App.tsx instead of each individual tab component. I also added some documentation for these patterns.

 As for the Firefox issue, it would seem this is related to OS settings: (https://stackoverflow.com/questions/11704828/how-to-allow-keyboard-focus-of-links-in-firefox/11713537#11713537) I confirmed this works, not sure of another workaround since using Link changes the Tab to be an anchor tag instead of a button.

File: packages/manager/src/features/linodes/LinodesDetail/LinodesDetailNavigation.tsx [hunk 3]
```diff
@@ -86,116 +80,62 @@ const LinodesDetailNavigation: React.StatelessComponent<CombinedProps> = props =
...
+            </TabPanel>
+            <TabPanel>
+              <VolumesLanding
+                linodeId={linodeId}
+                linodeLabel={linodeLabel}
+                linodeRegion={linodeRegion}
+                linodeConfigs={linodeConfigs}
+                readOnly={readOnly}
+                fromLinodes
+                removeBreadCrumb
```
Comment (johnwcallahan) [inline packages/manager/src/features/linodes/LinodesDetail/LinodesDetailNavigation.tsx:101]: There's a crashing bug if you go to the Linode Details page of a Linode with an attached Volume. It appears VolumesLanding is still looking for RouteProps here: https://github.com/linode/manager/blob/e006acaae577d2a5a3762ea0e1233708cc20984e/packages/manager/src/features/Volumes/VolumesLanding.tsx#L506

You should be able to change that to `const isVolumesLanding = this.props.linodeId === undefined;`, I believe.

**Comment by WilkinsKa1** [comment]
@johnwcallahan thanks for the review + feedback! I addressed everything except for the Safari issue- that appears to be related to an OS issue (I linked a related stack overflow link in the PR description) but you can fix this in your Safari preferences by checking the focus for all elements:
Screen Shot 2020-04-29 at 2 46 22 PM

I'm not sure if there's another way to handle this besides a hack with tabindex, which I'm hesitant to do as it would impact Reach's tabindex handling.

**Pull request merged**